transaction/core/src/validation/validate.rs

```diff
@@ -450,19 +450,26 @@ mod tests {
         mc_util_serial::deserialize(&bytes).unwrap()
     }
 
-    fn create_test_tx() -> (Tx, LedgerDB) {
+    fn create_test_tx(block_version: BlockVersion) -> (Tx, LedgerDB) {
         let mut rng: StdRng = SeedableRng::from_seed([1u8; 32]);
-        let sender = mc_transaction_core_test_utils::AccountKey::random(&mut rng);
+        let sender = AccountKey::random(&mut rng);
         let mut ledger = create_ledger();
         let n_blocks = 1;
-        initialize_ledger(&mut ledger, n_blocks, &sender, &mut rng);
+        initialize_ledger(
+            adapt_hack(&block_version),
```

Comment:
I would really like to refactor to kill off this `adapt_hack` stuff, I think we just have to reorganize the crates a little bit.